flake.nix

```diff
@@ -0,0 +1,41 @@
+{
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-24.11";
+    flake-utils.url = "github:numtide/flake-utils";
+    rust-overlay = {
+      url = "github:oxalica/rust-overlay";
+      inputs = {
+        nixpkgs.follows = "nixpkgs";
+      };
+    };
```

Comment:
Sorry, I'm not that familiar with nix - why is the overlay needed?

Reply:
It's a nice way to be able to select any Rust release that we may like, not just the one that's the default in a given nixpkgs release.